packages/firestore/src/local/indexeddb_mutation_queue.ts

```diff
@@ -184,80 +177,101 @@ export class IndexedDbMutationQueue implements MutationQueue {
   getNextBatchId(
     transaction: PersistenceTransaction
   ): PersistencePromise<BatchId> {
-    return PersistencePromise.resolve(this.nextBatchId);
+    let maxPendingBatchId = BATCHID_UNKNOWN;
+    const range = IDBKeyRange.lowerBound([this.userId]);
```

Comment:
Don't we need an upper bound to prevent iterating other users' batches?

FWIW- It's occurring to me that we could probably simplify things (and improve performance?) by switching to auto-generated IDs (this would also mean batchIDs would be unique across users which is not necessary but not harmful either).

Reply:
But since you're iterating backwards, this will always just read the last mutation stored, regardless of what user it's for, right?  So you need an upper bound to get to the right user...  I think.

But this all gets simpler if we just have global (rather than per-user) auto-incrementing ids.

Oh... I guess autoincrement may not work with compound keys...  We could potentially change the key to just be the batchId and then add an index on [userId, batchId] though.  Not sure if that's more or less expensive than just doing our own auto-incrementing id (i.e. have a mutationQueueGlobals store with a nextBatchId value that we increment every time we create a batch).